aspnetcore/mvc/controllers/filters.md

```diff
@@ -86,216 +82,224 @@ Several of the filter interfaces have corresponding attributes that can be used
 
-3. The Class filter `OnActionExecuting`
+You can set the `Order` property by using a constructor parameter:
 
-4. The Method filter `OnActionExecuting`
-
-5. The Method filter `OnActionExecuted`
+```csharp
+[MyFilter(Name = "Controller Level Attribute", Order=1)]
+```
 
-6. The Class filter `OnActionExecuted`
+If you have the same 3 Action filters shown in the preceding example but set the `Order` property of the controller and global filters to 1 and 2 respectively, the order of execution would be reversed.
 
-7. The Global filter `OnActionExecuted`
+| Sequence | Filter scope | `Order` property | Filter method |
+|:--------:|:------------:|:-----------------:|:-------------:|
+| 1 | Method | 0 | `OnActionExecuting` |
+| 2 | Controller | 1  | `OnActionExecuting` |
+| 3 | Global | 2  | `OnActionExecuting` |
+| 4 | Global | 2  | `OnActionExecuted` |
+| 5 | Controller | 1  | `OnActionExecuted` |
+| 6 | Method | 0  | `OnActionExecuted` |
 
-8. The Controller `OnActionExecuted`
+`Order` trumps scope when determining the order in which filters will run. Filters are sorted first by order, then scope is used to break ties. All of the built-in filters implement `IOrderedFilter` and set the default `Order` value to 0, so scope determines order unless you set `Order` to a non-zero value.
 
 > [!NOTE]
-> `IOrderedFilter` trumps scope when determining the order in which filters will run. Filters are sorted first by order, then scope is used to break ties. Order defaults to 0 if not set.
+> Every controller that inherits from the `Controller` base class includes `OnActionExecuting` and `OnActionExecuted` methods. These methods wrap the filters that run for a given action:  `OnActionExecuting` is called before any of the filters, and `OnActionExecuted` is called after all of the filters.
 
-To modify the default, scope-based order, you could explicitly set the `Order` property of a class-level or method-level filter. For example, adding `Order=-1` to a method level attribute:
-
-```csharp
-[MyFilter(Name = "Method Level Attribute", Order=-1)]
-   ```
-
-In this case, a value of less than zero would ensure this filter ran before both the Global and Class level filters (assuming their `Order` property was not set).
-
-The new order would be:
-
-1. The Controller `OnActionExecuting`
+## Authorization Filters
 
-2. The Method filter `OnActionExecuting`
+*Authorization Filters* control access to action methods and are the first filters to be executed within the filter pipeline. They have only a before method, unlike most filters that support before and after methods. You should only write a custom authorization filter if you are writing your own authorization framework. Note that you should not throw exceptions within authorization filters, since nothing will handle the exception (exception filters won't handle them). Instead, issue a challenge or find another way.
```

Comment:
I would add to this guidance:

`You should only write a custom authorization filter if you are writing your own authorization framework` 

prefer configuring your authorization policies or writing a custom authorization policy over writing a custom filter. Our filter implementation is just responsible for calling the authorization system.

This is based on us getting a lot of questions from users trying to do things the old way.